app/models/user.rb

```diff
@@ -3,13 +3,30 @@ class User < ActiveRecord::Base
   # :confirmable, :lockable, :timeoutable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :trackable, :validatable,
-         :omniauthable, omniauth_providers: [:twitter]
+         :omniauthable, omniauth_providers: [:twitter],
+         :authentication_keys => [:login]
+
   before_destroy :check_all_events_finished
 
   has_many :created_events, class_name: 'Event', foreign_key: :owner_id, dependent: :nullify
   has_many :tickets, dependent: :nullify
   has_many :participating_events, through: :tickets, source: :event
 
+  attr_accessor :login
+
+  def self.find_first_by_auth_conditions(warden_conditions)
+    conditions = warden_conditions.dup
+    if login = conditions.delete(:login)
```

Comment:
__[rubocop]__
* Lint/AssignmentInCondition: Assignment in condition - you probably meant to use `==`. [:ref](https://github.com/bbatsov/ruby-style-guide#safe-assignment-in-condition)